scaleway/docs-content
# feat: upgrade to MDX v2

## How to test
- Go on [this MR](https://github.com/scaleway/docs/pull/1480)
  - `nvm use && rm -rf ./node_modules && yarn && yarn clean`
  - `cd content && git checkout fix/mdxv2 && git pull origin fix/mdxv2`
  - `cd .. && yarn build`
- Make sure build is ok
- Check that pages are vastly ok
- Try to use the `--pages` flag. The flag let you now build complete subfolder or single page.
  - `yarn build --pages=compute` will build all the compute doc
  - `yarn build --pages=home` will build the homepage
  - `yarn build --pages=tutorials/xxxxxxx` will build only this specific page

## Wtf man?
Following the upgrade to MDX v2 on the source code, I did a pass on the syntax of your files. It's important to upgrade for the good health of the project!

## What's new

### **The bad news**
The only change that has a negative impact is concerning bullet listing (`ol-li`). The good syntax is now to have two tabs below each `li`. If you don't put those 2 tabs, the ordering will break (so instead of having `1. / 2. / 3.`, you will see `1. / 1. / 1.`). I did a pretty big check on a lit of files but of course I might not fix all of them. You'll be able to fix it on the go (please ? 🙏)

### **Functional change**
#### Blocks
Content in the `blocks/` folder are now `.json` files (instead of `.mdx`).

#### Frontmatter
`image` is replaced by `hero`. You need to specify the relative file of the asset 
**Example:**
```
---
...
hero: assets/general.webp
---
```

#### Lightbox
You don't need to specify the `assets/` in the `src` props of the component. Also, the system behind how we handle images have changed: it is highly recommended to uniquely name images.
**Example:**
```jsx
<Lightbox src="scaleway_xxxx_xxxx.webp" />
```

### **Some syntaxe update**
#### `<` character
You can't a `<` alone like this on your MDX. If you really want to display the `<` character, you must escape it with `\`
```
This is free for \< 1 hour
```

#### bullet list
Always add two tabs (4 spaces) below each `li` of the list. Also, don't put new line between each `li`.

Exemple:
Instead of
```jsx
1. xxxxxx
  I'm a string with two spaces before

2. xxxxxxx
I'm a string with no space before
3. xxxxxxxxx
  <Message ...>
 I'm content of a message!
  </Message>
```
do
```jsx
1. xxxxxx
    I'm a string with 4 spaces before
2. xxxxxxx
    I'm also a string with 4 spaces before
3. xxxxxxxxx
    <Message ...>
      I'm content of a message!
    </Message>
```


#### Message
It's not necessary to add new line after/before JSX tag. Also, add one tab (two spaces) for the content to show it's nested in it (like you would do in HTML)

Example:
Instead of 
```jsx
<Message type="note">

- xxxxxx
- xxxxxx
- xxxxxx
- xxxxxx

</Message>
```
do
```jsx
<Message type="note">
  - xxxxxx
  - xxxxxx
  - xxxxxx
  - xxxxxx
</Message>
````
